### ISSUE TITLE
🐛 Bug Report: [Accessibility] Component API  puzzle icon

### ISSUE THREAD

**Issue by DorisCCLin** [backstage/backstage#18696]
### 📜 Description

In our organization we use the ARC tool to test out A11y issues. 
In the Component API tab, on the table with the puzzle icon, there is no label for the button element here:

ARC toolkit description: No label for button element
Description: The <button> element lacks a programmatic name - text content, an explicit aria-label="...", or similar.

### 👍 Expected behavior

It should have have a button label for A11y purpose. 

### 👎 Actual Behavior with Screenshots

<img width="452" alt="API+definition+button" src="https://github.com/backstage/backstage/assets/27718001/79d96497-f321-4a20-9e97-0a155f890533">

### 👟 Reproduction steps

You should be able to see this bug when the data is in place: https://demo.backstage.io/catalog/default/component/playback-order/api

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by benjdlambert** [comment]
@DorisCCLin same for this one too, do you want to raise a PR with some suggested fixes? :pray:
